src/providers/ogr/qgsogrconnpool.h

```diff
@@ -61,7 +61,13 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     Q_OBJECT
 
   public:
-    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ) { initTimer( this ); }
+    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ), mRefCount( 0 ) { initTimer( this ); }
+    void ref() { ++mRefCount; }
```

Comment:
Really?  AFAICS the provider has a separate connection (r/w and not acquired from the connection pool) and only the feature iterators acquire connections from the connection pool.   Each connection should only be use by one feature iterator at a time.   So for parallel feature iterators there need to be separate connections - preferable unexpired connections that destroyed feature iterators returned to the pool or otherwise new ones.